trestle/tasks/osco_to_oscal.py

```diff
@@ -0,0 +1,184 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""OSCAL transformation tasks."""
+
+import configparser
+import logging
+import os
+import json
+import pathlib
+import traceback
+from typing import Optional
+
+import yaml
+
+from trestle.lib import osco
+from trestle.tasks.base_task import TaskBase
+from trestle.tasks.base_task import TaskOutcome
+
+logger = logging.getLogger(__name__)
+
+
+class OscoToOscal(TaskBase):
+    """
+    Task to convert OSCO yaml to OSCAL json.
+
+    Attributes:
+        name: Name of the task.
+    """
+
+    name = 'osco-to-oscal'
+
+    def __init__(self, config_object: Optional[configparser.SectionProxy]) -> None:
+        """
+        Initialize trestle task osco-to-oscal.
+
+        Attributes:
+            config_object: Config section associated with the task.
+        """
+        super().__init__(config_object)
+
+    def print_info(self) -> None:
+        """Print the help string."""
+        logger.info(f'Help information for {self.name} task.')
+        logger.info('Configuration flags sit under [task.osco-to-oscal].')
+        logger.info('input-dir = the path of the input directory comprising .yaml files.')
+        logger.info('output-dir = the path of the output directory comprising synthesized .oscal files.')
+        logger.info('output-overwrite = true [default] or false; replace existing output when true.')
+        logger.info('quiet = true or false [default]; display file creations and rules analysis when false.')
+
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        try:
+            if self._config:
+                idir = pathlib.Path(self._config.get('input-dir'))
+                if idir is None:
+                    logger.error(f'config missing "input-dir"')
+                    return TaskOutcome('simulated-failure')
+                odir = pathlib.Path(self._config.get('output-dir'))
+                if odir is None:
+                    logger.error(f'config missing "output-dir"')
+                    return TaskOutcome('simulated-failure')
+                overwrite = self._config.getboolean('output-overwrite', True)
+                for ifile in sorted(pathlib.Path(idir).iterdir()):
+                    parts = ifile.parts
+                    ifn = parts[len(parts)-1]
+                    if ifn.endswith('oscal-metadata.yaml'):
+                        continue
+                    if ifn.endswith('oscal-metadata.yml'):
+                        continue
+                    ofile = self._calculate_ofile(ifn, odir)
```

Comment:
The help information is not explicit about these particular files are required. If so we should state it in the help info print.

Reply:
Fixed. Added required/optional designation in the text.